app/code/core/Mage/Usa/Model/Shipping/Carrier/Ups.php

```diff
@@ -2341,6 +2341,10 @@ protected function _parseRestResponse(string $rateResponse)
             $rateResponseData = json_decode($rateResponse, true);
             if (@$rateResponseData['RateResponse']['Response']['ResponseStatus']['Description'] === 'Success') {
                 $arr = $rateResponseData['RateResponse']['RatedShipment'] ?? [];
+                if (isset($arr['Service'])) {
```

Comment:
```suggestion
                if (array_key_first($arr)) {
```

Reply:
Test: https://onlinephp.io?s=rZDRCoIwGIWvG-wdxghU6KZBXWTSRT1BXoqI6MQV6pgrCPHd2zTLYkYX7fKc8-_7z7_d8ZxDAMH8VFflEnnIavAxljT1c8YLWkq8QQ0EqHvYp-LKEjrWlLqvUi3h9QovRvKB1olgXLKq1C4erBaCtrXcJ5WYqcGQb_6EVdzwwe3YsRC6sF4hSmmifrP7KyyQFBfquH2GmDLklYHgGosovRTcVun4Fp3pLcqYqKXdIYKPZqGjp77OEPMMBCxDP0P0rWZmUzUKzE6oMO1bp6kfPA9NrNptegc%2C&v=8.0.30%2C7.4.33%2C8.3.0

Reply:
to me this is less readable

Reply:
Mhhh, "If first array-key is a string ..."

Still prefer my suggestion, but another one ...

`if (!array_key_exists(0, $arr)) {`

Reply:
Why do you mark it as resolved?

Reply:
because I prefer the code by @iamniels, easier to understand, also, 2 functions call instead of isset, there's no way it's better.

Reply:
`Isset` goes wrong in this case:

```
$json3 = '{"RatedShipment": [
      {"Service": null}
]}';
```
Better? You prefer it. May let others think about it, too? Nope?

Reply:
`isset` iterates over whole array, array_key_first does not. And is_string is not worth to mention.

Reply:
In most case 'isset' can be replaced by more strict checks .... however ...

Reply:
https://stackoverflow.com/a/9522522

isset is O1, provide better proves next time

Reply:
A note from @F1Red5 : https://github.com/php/php-src/pull/3360